File: mk_seeder.php
<?php

// Database connection settings
$db_host = 'localhost';
$db_user = 'root';
$db_password = '';
$db_name = 'old_learn_and_cure';

// Connect to the database (MySQL example)
$connection = new mysqli($db_host, $db_user, $db_password, $db_name);

// Check for connection errors
if ($connection->connect_error) {
    die("Connection failed: " . $connection->connect_error);
}

$tables = ['medical_subtopic_images','medical_subtopic_videos','medical_topics','medical_subjects','medical_subtopic_articles',
'subjects','vaccines',
'children_vaccines','excercises','excercise_images','medical_subjects','medical_subtopic_articles','medical_subtopic_images',
'bloodtests','bloodtest_details','bloodtest_names','bloodtest_records','bloodtest_units'
];
foreach ($tables as $tablekey => $tablevalue) {
    mk_seed($connection,$tablevalue);
}
// Close the database connection
$connection->close();

function mk_seed($connection,$table_name){
    // Define the table name and columns
    $table_name = $table_name;

    // Query the database to fetch data
    $query = "SELECT * FROM " . $table_name;
    $result = $connection->query($query);
    $className = toCamelCase($table_name);
    // Generate the Laravel seeder file
    $insert_array_string = "";
    $seeder_file_content = "<?php
    namespace Database\Seeders;

    use Illuminate\Database\Seeder;
    use Illuminate\Support\Facades\DB;


    class " . ucfirst($className) . "Seeder extends Seeder
    {
        public function run()
        {
    ";

    while ($row = $result->fetch_assoc()) {
        $insert_array_string .= makeArrayAsString($row).",";
    }
    $seeder_file_content .= "        DB::table('" . $table_name . "')->insert([".$insert_array_string. "]);\n";
    $seeder_file_content .= "    }
    }
    ";



    // Write the seeder file
    $seeder_file_name = $className . "Seeder.php";
    file_put_contents($seeder_file_name, $seeder_file_content);

    echo "Seeder file '{$seeder_file_name}' has been generated.\n";
}


function makeArrayAsString($array){
    $output = '';
    if(count($array)>0)
    {
        $output .= '[';
        foreach ($array as $key => $value) {
            if(!empty($value))
            {
                $output .= "'$key' => '$value' ,";
            }
            else
            {
                $output .= "'$key' => null ,";
            }
        }
        $output .= "]";
    }
    return $output;
}

function toCamelCase($value)
{
    // Split the string by underscores
    $parts = explode('_', $value);

    // Capitalize the first letter of each part and join them
    $camelCaseValue = implode('', array_map('ucfirst', $parts));

    return $camelCaseValue;
}

File: mk_seeder.py
import pymysql  # Assuming you're using MySQL. You can change this library based on your SQL database.

# Database connection settings
db_host = 'localhost'
db_user = 'root'
db_password = ''
db_name = 'old_learn_and_cure'

# Connect to the database
connection = pymysql.connect(host=db_host, user=db_user, password=db_password, database=db_name)

# Define the table name and columns
table_name = 'vaccines'
columns = ['column1', 'column2', 'column3']  # Replace with the actual column names

# Query the database to fetch data
cursor = connection.cursor()
query = f"SELECT * FROM {table_name}"
cursor.execute(query)
data = cursor.fetchall()

# Close the database connection
connection.close()

# Generate the Laravel seeder file
seeder_file_content = """<?php

use Illuminate\Database\Seeder;
use App\Models\YourModel;  // Replace with your actual model

class YourTableSeeder extends Seeder
{
    public function run()
    {
"""

for row in data:
    seeder_file_content += f"        YourModel::create({str(row)});\n"  # Replace with your actual model

seeder_file_content += """    }
}
"""

# Write the seeder file
seeder_file_name = f"{table_name}_seeder.php"
with open(seeder_file_name, 'w') as seeder_file:
    seeder_file.write(seeder_file_content)

print(f"Seeder file '{seeder_file_name}' has been generated.")
